.changeset/shiny-swans-relax.md

```diff
@@ -0,0 +1,123 @@
+---
+"fluid-framework": minor
+"@fluidframework/tree": minor
+---
+---
+"section": tree
+---
+
+Improve typing when exact TypeScript type for a schema is not provided
+
+The Tree APIs are designed to be used in a strongly typed way, with the full TypeScript type for the schema always being provided.
+Due to limitations of the TypeScript language, there was not a practical way to prevent less descriptive types, like `TreeNodeSchema` or `ImplicitFieldSchema` from being used where the type of a specific schema was intended.
+Code which does this will encounter several issues with tree APIs, and this change fixes some of those issues.
+This change  mainly fixes that `NodeFromSchema<TreeNodeSchema>` used to return `unknown` and now returns `TreeNode | TreeLeafValue`.
+
+This change by itself seems mostly harmless, as it just improves the precision of the typing in this one edge case.
+Unfortunately there are other typing bugs which complicate the situation, resulting in APIs for inserting data into the tree to also behave poorly when given non-specific types like `TreeNodeSchema`.
+These APIs include cases like `TreeView.initialize`.
+
+This incorrectly allowed some usage like taking a type erased schema and initial tree pair, and creating a view of type `TreeView<ImplicitFieldSchema>` then initializing a it.
+With the typing being partly fixed, some unsafe inputs are still allowed when trying to initialize such a view, but some are now prevented.
+
+While this use-case of modifying in code not strongly typed by the exact schema, was not intended to be supported,
```

Comment:
This clause is a bit hard to parse, I think there might be a typo here, but I'm not sure what the intention was.